src/datatype/aggregate_svecf32.rs

```diff
@@ -0,0 +1,792 @@
+                (s1, s2, 0)
+            } else {
+                (s2, s1, 1)
+            };
+            let total_count = s1.count() + s2.count();
+            match s1.check_capacity(s2.len()) {
+                true => {
+                    s1.merge_in_place(SVecf32Borrowed::new(
+                        s2.dims() as u32,
+                        s2.indexes(),
+                        s2.values(),
+                    ));
+                    s1.count = total_count;
+                    if larger_internal == 0 {
+                        state1
+                    } else {
+                        state2
+                    }
+                }
+                false => {
+                    let mut new_state = SVecf32AggregateAvgSumStype::new_with_capacity(
+                        dims1 as u32,
+                        s1.len() + s2.len(),
+                    );
+                    new_state.merge_in_place(SVecf32Borrowed::new(
+                        dims1 as u32,
+                        s1.indexes(),
+                        s1.values(),
+                    ));
+                    new_state.merge_in_place(SVecf32Borrowed::new(
+                        s2.dims() as u32,
+                        s2.indexes(),
+                        s2.values(),
+                    ));
+                    new_state.count = total_count;
+                    Internal::new(new_state)
+                }
+            }
+        }
+    }
+}
+#[no_mangle]
+#[doc(hidden)]
+pub unsafe extern "C" fn _vectors_svecf32_aggregate_avg_sum_combinefunc_wrapper(
+    _fcinfo: ::pgrx::pg_sys::FunctionCallInfo,
+) -> ::pgrx::pg_sys::Datum {
+    #[allow(non_snake_case)]
+    unsafe fn _vectors_svecf32_aggregate_avg_sum_combinefunc_wrapper_inner(
+        _fcinfo: ::pgrx::pg_sys::FunctionCallInfo,
+    ) -> ::pgrx::pg_sys::Datum {
+        let state1_ = unsafe {
+            ::pgrx::fcinfo::pg_getarg::<Internal>(_fcinfo, 0usize).unwrap_or_else(|| {
+                panic!(
+                    "{} is null",
+                    stringify! {
+                      state1_
+                    }
+                )
+            })
+        };
+        let state2_ = unsafe {
+            ::pgrx::fcinfo::pg_getarg::<Internal>(_fcinfo, 1usize).unwrap_or_else(|| {
+                panic!(
+                    "{} is null",
+                    stringify! {
+                      state2_
+                    }
+                )
+            })
+        };
+        #[allow(unused_unsafe)]
+        let result = unsafe { _vectors_svecf32_aggregate_avg_sum_combinefunc(state1_, state2_) };
+        ::pgrx::datum::IntoDatum::into_datum(result)
+            .unwrap_or_else(|| panic!("returned Datum was NULL"))
+    }
+    #[allow(unused_unsafe)]
+    unsafe {
+        pgrx::pg_sys::submodules::panic::pgrx_extern_c_guard(move || {
+            let mut agg_context: *mut ::pgrx::pg_sys::MemoryContextData = std::ptr::null_mut();
```

Comment:
pg will switch to a temporary memory context when call aggregate trans function.
https://github.com/postgres/postgres/blob/52b49b796cc7fd976f4da6aa49c9679ecdae8bd5/src/backend/executor/nodeAgg.c#L761-L801
[doxygen](https://doxygen.postgresql.org/nodeAgg_8c.html#a8756447ac100aaea4999db35c4280939)
So we need to switch context like https://github.com/postgres/postgres/blob/7c655a04a2dc84b59ed6dce97bd38b79e734ecca/src/backend/utils/adt/numeric.c#L5635-L5648.
Currently pgrx's `Aggregate` trait can't pass a reference with lifetime. https://github.com/pgcentralfoundation/pgrx/blob/develop/pgrx/src/aggregate.rs. So I expand it. （ Of course, this piece of code is terrible.）

Reply:
https://github.com/pgcentralfoundation/pgrx/issues/1719

Reply:
I feel not necessary for using `Aggregate` trait.

Reply:
I don't find a good way to swtich the context under pgrx currently.😭

Reply:
You could just write code in C way. `pgrx` is more a binding than a complete interface of PostgreSQL.

Reply:
> You could just write code in C way. `pgrx` is more a binding than a complete interface of PostgreSQL.

I add a macro to wrap it currently.